tests/compiler/bool.ts

```diff
@@ -6,9 +6,52 @@ var u = <u32>2;
 assert(<bool>u == true);
 var U = <u64>2;
 assert(<bool>U == true);
+
 var f = <f32>2;
 assert(<bool>f == true);
+var f0 = <f32>+0.0;
+assert(<bool>f0 == false);
+var f1 = <f32>-0.0;
+assert(<bool>f1 == false);
+var f2 = <f32>+NaN;
+assert(<bool>f2 == false);
+var f3 = <f32>-NaN;
+assert(<bool>f3 == false);
```

Comment:
Particularly interesting cases are `NaN`s with non-canonicalized bitpatterns, created by means of `reinterpret`.

Reply:
non-canonicalized NaN should also handled. However not sure `-NaN` produce non-canonical form, so test may be unnecessary

Reply:
Ah, It should!) Only `NaN * -1` or `0 - NaN` do canonization, but `(neg x)` should just change bit regardless of values

Reply:
Would however prefer to add the four reinterpreted minimum/maximum quiet/silent NaNs just to be sure :)